modules/subchannel/src/problems/SubChannel1PhaseProblem.C

```diff
@@ -0,0 +1,2928 @@
+    {
+      int last_level = (iblock + 1) * _block_size;
+      int first_level = iblock * _block_size + 1;
+      Real T_block_error = 1.0;
+      auto T_it = 0;
+      _console << "Solving Block: " << iblock << " From first level: " << first_level
+               << " to last level: " << last_level << std::endl;
+      while (T_block_error > _T_tol && T_it < _T_maxit)
+      {
+        T_it += 1;
+        if (T_it == _T_maxit)
+        {
+          _console << "Reached maximum number of temperature iterations for block: " << iblock
+                   << std::endl;
+          _converged = false;
+        }
+        auto T_L2norm_old_block = _T_soln->L2norm();
+
+        if (_segregated_bool)
+        {
+          computeWijFromSolve(iblock);
+
+          if (_compute_power)
+          {
+            computeh(iblock);
+            computeT(iblock);
+          }
+        }
+        else
+        {
+          LibmeshPetscCall(implicitPetscSolve(iblock));
+          computeWijPrime(iblock);
+          if (_verbose_subchannel)
+            _console << "Done with main solve." << std::endl;
+          if (_monolithic_thermal_bool)
+          {
+            // Enthalpy is already solved from the monolithic solve
+            computeT(iblock);
+          }
+          else
+          {
+            if (_verbose_subchannel)
+              _console << "Starting thermal solve." << std::endl;
+            if (_compute_power)
+            {
+              computeh(iblock);
+              computeT(iblock);
+            }
+            if (_verbose_subchannel)
+              _console << "Done with thermal solve." << std::endl;
+          }
+        }
+
+        if (_verbose_subchannel)
+          _console << "Start updating thermophysical properties." << std::endl;
+
+        if (_compute_density)
+          computeRho(iblock);
+
+        if (_compute_viscosity)
+          computeMu(iblock);
+
+        if (_verbose_subchannel)
+          _console << "Done updating thermophysical properties." << std::endl;
+
+        // We must do a global assembly to make sure data is parallel consistent before we do things
+        // like compute L2 norms
+        _aux->solution().close();
+
+        auto T_L2norm_new = _T_soln->L2norm();
+        T_block_error =
+            std::abs((T_L2norm_new - T_L2norm_old_block) / (T_L2norm_old_block + 1E-14));
+        _console << "T_block_error: " << T_block_error << std::endl;
+      }
+    }
+    auto P_L2norm_new_axial = _P_soln->L2norm();
+    P_error =
+        std::abs((P_L2norm_new_axial - P_L2norm_old_axial) / (P_L2norm_old_axial + _P_out + 1E-14));
+    _console << "P_error :" << P_error << std::endl;
+    if (_verbose_subchannel)
+    {
+      _console << "Iteration:  " << P_it << std::endl;
+      _console << "Maximum iterations: " << P_it_max << std::endl;
+    }
+  }
+  // update old crossflow matrix
+  _Wij_old = _Wij;
+  _console << "Finished executing subchannel solver\n";
+
```

Comment:
I think we should check here, at convergence, that the cross flow resistance term did go to zero as desired

Reply:
the basis is that adding resistance to cross flows makes convergence easier, but likely biases the solution.
If there is a good reference for keeping a non-zero cross flow resistance term, we can instead point to that in the documentation

Reply:
Yes, our multiple papers showing the validation for subchannel. We won't be doing this as I deem is not necessary unless proven otherwise

Reply:
If the claim is that the term should go to zero when the solution converges, then I think the check is important, as it would indicate a significant bug.

If there is no claim that it should go to zero, but instead it's known to be artificial but beneficial to convergence, but deemed an acceptable error based on validation results, that's ok too - we just need to document it as such for clarity.

Reply:
please document this

Reply:
See issue #30037